lib/runa/client.rb

```diff
@@ -32,10 +30,17 @@ def initialize(options = {})
       end
     end
 
-    def request(method, path, payload = {})
+    def request(method, path, payload = {}, key)
       @connection.send(method) do |req|
         req.url [@api_path, path].join
-        req.headers['Content-Type'] = 'application/json'
+        req.headers['Content-Type'] = 'application/json'.freeze
+        req.headers['X-Api-Key'] = @api_key
+        req.headers['X-Api-Version'] = '2024-02-05'.freeze
+        req.headers['X-Execution-Mode'] = 'sync'.freeze
+        if method.eql? :post
+          raise StandardError('Required idempotency key missing') if key.eql? ''
```

Comment:
```suggestion
          raise StandardError('Required idempotency key missing') if key.blank? ''
```

`.blank?` is better: https://blog.arkency.com/2017/07/nil-empty-blank-ruby-rails-difference/

Reply:
I'll try it again but it kept giving me an error 😅

Reply:
```suggestion
          raise StandardError('Required idempotency key missing') if key.nil? || key.strip == ''
```